templates-pipelines/windows-efi-installer/README.md

```diff
@@ -0,0 +1,93 @@
+   Different autounattend.xml can be also passed in a separate ConfigMap with the Pipeline parameter `autounattendConfigMapName` when creating a PipelineRun.
+
+## Pipeline Description
+
+```
+  import-win-iso --- modify-windows-iso-file --- create-vm --- wait-for-vmi-status --- create-base-dv --- cleanup-vm
+                                              |
+                       create-vm-root-disk --- 
+```
+
+1. `create-vm-root-disk` creates empty DV which is used for windows installation.
+2. `import-win-iso` creates new datavolume with windows iso file with name defined in `isoDVName` parameter.
+3. `modify-windows-iso-file` extracts imported iso, replaces prompt bootloader (which is used as a default one when EFI is used) 
+   with no-prompt bootloader, pack the updated files back to new iso, convert the iso and replaces original iso with updated one.
+   Replacement of bootloader is needed to be able to automate installation of windows versions which require EFI.
+4. `create-vm` task creates a VM.
+   A DV with the Windows source ISO will be attached as CD-ROM and a second empty DV will be used as installation destination.
+   A third DV with the virtio-win ISO will also be attached. (Pipeline parameter `virtioContainerDiskName`)
+5. `wait-for-vmi-status` task waits until the VM shuts down.
+6. `create-base-dv` task creates an DV with the specified name and namespace (Pipeline parameters `baseDvName` and 
+   `baseDvNamespace`).
+   Then it clones the second DV of the installation VM into the new DV.
+7. `cleanup-vm` deletes the installer VM and all of its DVs.
+8. The output artifact will be the `baseDvName`/`baseDvNamespace` DV with the basic Windows installation.
+   It will boot into the Windows OOBE and needs to be setup further before it can be used.
+
+## How to run
+
+Pipeline runs with resolvers:
+```yaml
+{% for item in pipeline_runs_yaml %}
+{{ item | to_nice_yaml }}---
+{% endfor %}
+```
+
+## Cancelling/Deleteting pipelineRuns
+
+When running the example pipelines, they create temporary objects (DataVolumes, VMs, templates, ...). Each pipeline has its own clean up system which 
+should keep the cluster clean from leftovers. In case user hard deletes or cancels running pipelineRun, the pipelineRun will not clean temporary 
+objects and objects will stay in the cluster and then they have to be deleted manually. To prevent this behaviour, cancel the 
+[pipelineRun gracefully](https://tekton.dev/docs/pipelines/pipelineruns/#gracefully-cancelling-a-pipelinerun). It triggers special tasks, 
+which remove temporary objects and keep only result DataVolume/PVC.
+
+windows-efi-installer pipeline generates for each pipelineRun new source datavolume which contains imported iso. This DV has generated name and is 
+deleted after pipeline succeeds. However, the created PVC will stay in cluster, but it will have terminating state. It will wait, until pipelinRun is 
```

Comment:
```suggestion
deleted after pipeline succeeds. However, the created PVC will stay in cluster, but it will have terminating state. It will wait until pipelinRun is 
```